sympy/matrices/expressions/matexpr.py

```diff
@@ -110,7 +110,11 @@ def __rmul__(self, other):
     def __pow__(self, other):
         if not self.is_square:
             raise ShapeError("Power of non-square matrix %s" % self)
-        if other is S.NegativeOne:
+        if self.is_ZeroMatrix:
+            return self
```

Comment:
Yes.
If an instance of a zero matrix is raised to -ve power it will raise the value error saying matrix is invertible, as in : (ZeroMatrix(3,3))**-2